catalog/MDCCatalog/MDCNodeListViewController.swift

```diff
@@ -382,26 +382,32 @@ extension MDCNodeListViewController {
         container.appBar.navigationBar.titleTextAttributes =
             [ NSForegroundColorAttributeName: UIColor.white, NSFontAttributeName: appBarFont ]
 
-        // TODO(featherless): Remove once
-        // https://github.com/material-components/material-components-ios/issues/367 is resolved.
-        contentVC.title = node.title
-
-        let headerView = container.appBar.headerViewController.headerView
-
         let appDelegate = UIApplication.shared.delegate as! AppDelegate
         let colorScheme = appDelegate.colorScheme
         MDCFlexibleHeaderColorThemer.apply(colorScheme, to: MDCFlexibleHeaderView.appearance())
-
         let textColor = UIColor.white
         UIBarButtonItem.appearance().setTitleTextAttributes(
           [NSForegroundColorAttributeName: textColor],
           for: UIControlState())
 
-        var contentFrame = container.contentViewController.view.frame
-        let headerSize = headerView.sizeThatFits(container.contentViewController.view.frame.size)
-        contentFrame.origin.y = headerSize.height
-        contentFrame.size.height = self.view.bounds.size.height - headerSize.height
-        container.contentViewController.view.frame = contentFrame
+        // TODO(featherless): Remove once
+        // https://github.com/material-components/material-components-ios/issues/367 is resolved.
+        contentVC.title = node.title
+
+        let headerView = container.appBar.headerViewController.headerView
+        if contentVC.isKind(of: MDCCollectionViewController.self) {
+          let collectionVC = contentVC as! MDCCollectionViewController
+          headerView.trackingScrollView = collectionVC.collectionView
+        } else if contentVC.view.isKind(of: UIScrollView.self) {
+          headerView.trackingScrollView = contentVC.view as? UIScrollView
+        } else {
+          // TODO(chuga): This is bad. We should be adjusting for Safe Area changes.
+          var contentFrame = container.contentViewController.view.frame
+          let headerSize = headerView.sizeThatFits(container.contentViewController.view.frame.size)
+          contentFrame.origin.y = headerSize.height
+          contentFrame.size.height = self.view.bounds.size.height - headerSize.height
```

Comment:
You don't have to go thru `size` to get to height. You can `rect.height` in swift.